src/frontends/jax/src/frontend.cpp

```diff
@@ -0,0 +1,249 @@
+    {
+        TranslateSession translate_session(model, supported_ops, m_telemetry);
+        partial_model = translate_session.get_converted_model();
+    }
+    try {
+        normalize(partial_model);
+    } catch (...) {
+        // normalize can fail on transformation, but the model is still valid. We can return such model.
+        // If model can be validated we suppress normalize exception.
+        partial_model->validate_nodes_and_infer_types();
+    }
+    return partial_model;
+}
+
+std::shared_ptr<Model> FrontEnd::decode(const InputModel::Ptr& model) const {
+    FRONT_END_NOT_IMPLEMENTED(decode);
+}
+
+void FrontEnd::normalize(const std::shared_ptr<ov::Model>& model) const {
+    ov::pass::Manager manager;
+
+    // the following 2 transformations are needed for keypoint detectron2 models to work.
+    // AtenIndexToSelect will be called twice
+    manager.register_pass<ov::pass::ConvertConvertLike>();
+
+    // Mark quantized and f16/bf16 compressed constants to prevent CF for them,
+    // so that not extra memory is used for intermediate decompressed constants.
+    manager.register_pass<ov::pass::MarkDequantizationSubgraph>(
+        element::TypeVector{element::u8, element::i8, element::u4, element::i4});
+    manager.register_pass<ov::pass::MarkCompressedFloatConstants>();
+    manager.register_pass<ov::pass::ConstantFolding>();
+
+    manager.register_pass<ov::pass::ConvertConvertPromoteTypes>();
+    manager.register_pass<ov::pass::PushConstantToSubgraph>();
+    manager.register_pass<ov::pass::UnrollIf>();
+    // TODO: this is only a fake pass and needs to be removed.
+    manager.register_pass<ov::frontend::jax::pass::PlaceholderInBodyReplacer>();
+    manager.register_pass<ov::pass::RemoveMultiSubGraphOpDanglingParamsResults>();
+    manager.register_pass<ov::pass::ReverseShapeAndTypeInfer>();
+    manager.register_pass<ov::pass::ResolveNameCollisions>(true);
+    manager.run_passes(model);
+
+    // Usually if nn.Module.forward is given as a source model for conversion, there is the first Parameter
+    // that represents original `self` argument in forward(self, ...). `self` shouldn't play any role in model
+    // inference if model is completely frozen and all methods are inlined. So we check if it doesn't have any
+    // consumers in the finally converted model and remove this parameter. This parameter should have index 0.
+    if (model->get_parameters().size() > 0) {
+        auto self = model->get_parameters()[0];
+        if (self->output(0).get_target_inputs().empty()) {
+            // There is no consumers: safe to remove
+            OPENVINO_DEBUG << "[ WARNING ] Removing parameter[0] in converted jax model, because it is never used "
+                              "and treated as `self`\n";
+            model->remove_parameter(self);
+        } else {
+            OPENVINO_DEBUG << "[ WARNING ] Couldn't remove parameter[0] in converted jax model\n";
+        }
+    }
```

Comment:
Remove all the passes for now, they are not required in general and should be added if needed